src/intro-to-c/arrays/arrays-intro.md

```diff
@@ -0,0 +1,300 @@
+
+```
++------+------+------+------+------+
+|      |      |      |      |      |
++------+------+------+------+------+
+```
+
+It's empty and barren. Let's put some grades into it!
+
+To access each cell in an array, we have to know what _index_ we want to access first. The index represents the location of a cell in an array. An important thing to note about C (and many other programming languages) is that array indexes start at **0**.
+
+What does this mean? Well, if we wanted the index of the first cell, we wouldn't use 1. Instead, we would use 0 to represent the first cell in an array. This also means that if we wanted to access the last cell in an array, we would use `n - 1`, `n` being the size of the array. For instance, in our `grades` array of size 5, if we wanted to access the last element, we would use index 4, since we start from 0 for our indexes.
+
+To access an index in an array, we can use brackets (`[]`).
+Here, let's access the first cell of `grades`, and assign the first cell to `92.5`:
+
+```c
+grades[0] = 92.5; // first cell of the grades array will have an index of 0
+```
+
+Great! Now we have one grade in... but what about the rest?
+Our array currently looks like this now:
+
+```
++------+------+------+------+------+
+| 92.5 |      |      |      |      |
++------+------+------+------+------+
+```
+
+We've populated the first cell with a grade, by using our array access operator. We can do this for the rest of the grades as well:
+
+```c
+grades[0] = 92.5;
+grades[1] = 52.4;
+grades[2] = 75.0;
+grades[3] = 82.0;
+grades[4] = 88.9; // this is the last element we can access in our grades array!
+```
+
+Here is what our array looks like now:
+
+```
++------+------+------+------+------+
+| 92.5 | 52.4 | 75.0 | 82.0 | 88.9 |
++------+------+------+------+------+
+```
+
+You can also overwrite the values in an array by using the array access operator:
+```c
+grades[3] = 23.0 // they failed...
+```
+
+Here's a preview of the array after overwriting what's in index 3 of `grades`:
+```
++------+------+------+------+------+
+| 92.5 | 52.4 | 75.0 | 23.0 | 88.9 |
++------+------+------+------+------+
+```
+
+Now we finally have our array full of grades! Obviously, this is a pretty tedious way to populate our array; we just manually typed out the grades for each student! However, there are other ways in which we can initialize and populate an array:
+
+## Other Ways to Create and Populate an Array
+
+### Initializer List
+If we already know what will be in our array, we can create an array like this, using what is called an initializer list:
+```c
+int favorite_numbers[] = {3, 5, 9, 42, 1, 3141592};
+```
+Notice how we didn't need to specify the size of the array. This is because we used curly brackets (`{}`) to 'initialize' our array inline, so C will automagically create our array with the smallest size needed to house all of our elements. We can verify this by using `sizeof`:
+
+<details>
+<summary>Show code</summary>
```

Comment:
Cool use of the `<details>` tag! This may play a little strangely with other export formats, but I do not see a huge issue with it for now ;)